Formula/jpeg-xl.rb

```diff
@@ -0,0 +1,50 @@
+class JpegXl < Formula
+  desc "New file format for still image compression"
+  homepage "https://jpeg.org/jpegxl/index.html"
+  url "https://gitlab.com/wg1/jpeg-xl/-/archive/v0.2/jpeg-xl-v0.2.tar.bz2"
+  sha256 "f0933c796f95ee905efa7a677367c0d57678b9587c2e967ea30d72e9405cca72"
+  license "Apache-2.0"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "brotli"
+  depends_on "giflib"
+  depends_on "ilmbase"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "openexr"
+  depends_on "webp"
+
+  # These resources are versioned according to the script supplied with jpeg-xl to download the dependencies:
+  # https://gitlab.com/wg1/jpeg-xl/-/blob/master/deps.sh
+  resource "highway" do
+    url "https://github.com/google/highway.git",
+        revision: "311c183c9d96e69b123f61eedc21025dd27be000"
+  end
+  resource "lodepng" do
+    url "https://github.com/lvandeve/lodepng.git",
+        revision: "48e5364ef48ec2408f44c727657ac1b6703185f8"
+  end
+  resource "sjpeg" do
+    url "https://github.com/webmproject/sjpeg.git",
+        revision: "868ab558fad70fcbe8863ba4e85179eeb81cc840"
+  end
+  resource "skcms" do
+    url "https://skia.googlesource.com/skcms.git",
+        revision: "64374756e03700d649f897dbd98c95e78c30c7da"
+  end
+
+  def install
+    resources.each { |r| r.stage buildpath/"third_party"/r.name }
+    mkdir "build" do
+      system "cmake", "..", "-DBUILD_TESTING=OFF", *std_cmake_args
+      system "cmake", "--build", "."
+      system "cmake", "--build", ".", "--target", "install"
+    end
+  end
+
+  test do
+    system "cjxl", test_fixtures("test.jpg"), "test.jxl"
```

Comment:
Actually, since there are outstanding changes, may as well:
```suggestion
    system "#{bin}/cjxl", test_fixtures("test.jpg"), "test.jxl"
```
to make sure we're not picking up some other binary that happened to also be called `cjxl`